YARG.Core/MoonscraperChartParser/IO/Chart/ChartReader.ProcessLists.cs

```diff
@@ -201,31 +201,40 @@ private static List<NoteEventProcessFn> GetInitialPostProcessList(MoonChart.Game
             };
         }
 
-        private class SoloPhraseConverter : ITextPhraseConverter
+        private static unsafe void ConvertSoloEvents(ref NoteProcessParams noteProcessParams)
         {
-            private readonly MoonChart _chart;
-
-            public string StartEvent => TextEvents.SOLO_START;
-            public string EndEvent => TextEvents.SOLO_END;
-
-            public SoloPhraseConverter(MoonChart chart)
-            {
-                _chart = chart;
-            }
-
-            public void AddPhrase(uint startTick, uint endTick)
+            var chart = noteProcessParams.chart;
+            // Keeps tracks of soloes that start on the same tick when another solo ends
+            var soloQueue = stackalloc uint[2] { uint.MaxValue, uint.MaxValue };
```

Comment:
No real need to stackalloc here if you're not using this as an actual array or buffer, it could just as easily be two separate variables. I feel the intent becomes obscured otherwise:
```suggestion
            uint startTick = uint.MaxValue;
            uint nextStartTick = uint.MaxValue;
```